client/src/components/CurrentExerciseCard.tsx

```diff
@@ -0,0 +1,138 @@
+import React from "react";
+import Box from "@mui/material/Box";
+import Typography from "@mui/material/Typography";
+import ArrowBackIcon from '@mui/icons-material/ArrowBack';
+import ArrowForwardIcon from '@mui/icons-material/ArrowForward';
+import CurrentExerciseStateIcons, { CurrentExerciseStateIconsProps, ExerciseSetState} from "./CurrentExerciseStateIcons.tsx";
+import Button from "@mui/material/Button";
+import CompleteSetButton from "./CompleteSetButton.tsx";
+
+type ExerciseSetProps = {
+  id: number;
+  weight: number;
+  reps: number;
+  completedAt?: Date;
+}
+
+type CurrentWorkoutCardProps = {
+  id: number;
+  name: string;
+  exerciseSets: ExerciseSetProps[];
+}
+
+export default function CurrentWorkoutCard
+  ({ id, name, exerciseSets }: CurrentWorkoutCardProps
+) {
+
+  function getLatestIncompleteExerciseSet(): ExerciseSetProps | null {
+    for (let i = exerciseSets.length - 1; i >= 0; i--) {
+      if (!exerciseSets[i].completedAt) {
+        return exerciseSets[i];
+      }
+    }
+
+    return null;
+  }
+
```

Comment:
The implementation of `getIconStates()` will change to use data coming from the API once it's available.